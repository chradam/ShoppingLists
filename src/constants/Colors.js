export default {
  primary: '#1f145c',
};
