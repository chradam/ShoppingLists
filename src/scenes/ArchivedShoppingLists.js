import React, {useState, useCallback} from 'react';
import {
  View,
  Text,
  Animated,
  TouchableHighlight,
  TouchableOpacity,
  StatusBar,
} from 'react-native';
import {SwipeListView} from 'react-native-swipe-list-view';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import {useFocusEffect} from '@react-navigation/native';

import {listsStyles} from '../styles';
import {LIST_MODE, LIST_STATUS} from '../config/Settings';
import model from '../model/Model';
import {sortList} from '../utils';
import {SortListsSwitch} from '../components/SortListsSwitch';

export const ArchivedShoppingLists = ({navigation}) => {
  const [archivedShoppingLists, setArchivedShoppingLists] = useState(null);
  const [sortType, setSortType] = useState('desc');

  useFocusEffect(
    useCallback(() => {
      model.readShoppingLists(LIST_STATUS.archived).then(lists => {
        const sortedLists = sortList(lists, sortType);

        setArchivedShoppingLists(
          sortedLists.map((list, index) => ({
            key: `${index}`,
            id: list.id,
            date: list.date,
            title: list.title,
            products: list.products,
            status: list.status,
          })),
        );
      });

      return () => {};
    }, [sortType]),
  );

  const closeRow = (rowMap, rowKey) => {
    if (rowMap[rowKey]) {
      rowMap[rowKey].closeRow();
    }
  };

  const unarchiveRow = async (rowMap, rowKey, list) => {
    closeRow(rowMap, rowKey);
    const newData = [...archivedShoppingLists];
    const prevIndex = archivedShoppingLists.findIndex(
      item => item.key === rowKey,
    );
    newData.splice(prevIndex, 1);
    setArchivedShoppingLists(newData);

    model.updateShoppingList(list, LIST_STATUS.active);
  };

  const deleteRow = (rowMap, rowKey, itemId) => {
    model
      .deleteShoppingList(itemId)
      .then(() => {
        closeRow(rowMap, rowKey);
        const newData = [...archivedShoppingLists];
        const prevIndex = archivedShoppingLists.findIndex(
          item => item.key === rowKey,
        );
        newData.splice(prevIndex, 1);
        setArchivedShoppingLists(newData);
      })
      .catch(err => {
        console.log(err);
      });
  };

  const onRowDidOpen = rowKey => {
    // console.log('This row opened', rowKey);
  };

  const onLeftActionStatusChange = rowKey => {
    console.log('onLeftActionStatusChange', rowKey);
  };

  const onRightActionStatusChange = rowKey => {
    console.log('onRightActionStatusChange', rowKey);
  };

  const onRightAction = rowKey => {
    console.log('onRightAction', rowKey);
  };

  const onLeftAction = rowKey => {
    console.log('onLeftAction', rowKey);
  };

  const VisibleItem = props => {
    const {
      data,
      rowHeightAnimatedValue,
      removeRow,
      leftActionState,
      rightActionState,
    } = props;

    if (rightActionState) {
      Animated.timing(rowHeightAnimatedValue, {
        toValue: 0,
        duration: 200,
        useNativeDriver: false,
      }).start(() => {
        removeRow();
      });
    }

    return (
      <Animated.View
        style={[listsStyles.rowFront, {height: rowHeightAnimatedValue}]}>
        <TouchableHighlight
          style={listsStyles.rowFrontVisible}
          onPress={() =>
            navigation.navigate('AddShoppingList', {
              name: 'Archived Shopping List',
              mode: LIST_MODE.archived,
              paramList: data.item,
            })
          }
          underlayColor={'#aaa'}>
          <View>
            <Text style={listsStyles.title} numberOfLines={1}>
              {data.item.title}
            </Text>
            <Text>
              {new Date(data.item.date).toLocaleString('pl-PL', {
                timeZone: 'Europe/Warsaw',
              })}
            </Text>
          </View>
        </TouchableHighlight>
      </Animated.View>
    );
  };

  const renderItem = (data, rowMap) => {
    const rowHeightAnimatedValue = new Animated.Value(60);

    return (
      <VisibleItem
        data={data}
        rowHeightAnimatedValue={rowHeightAnimatedValue}
        removeRow={() => deleteRow(rowMap, data.item.key, data.item.id)}
      />
    );
  };

  const HiddenItemWithActions = props => {
    const {
      swipeAnimatedValue,
      leftActionActivated,
      rightActionActivated,
      rowActionAnimatedValue,
      rowHeightAnimatedValue,
      onClose,
      onDelete,
    } = props;

    if (rightActionActivated) {
      Animated.spring(rowActionAnimatedValue, {
        toValue: 500,
        useNativeDriver: false,
      }).start();
    } else {
      Animated.spring(rowActionAnimatedValue, {
        toValue: 75,
        useNativeDriver: false,
      }).start();
    }

    return (
      <Animated.View
        style={[listsStyles.rowBack, {height: rowHeightAnimatedValue}]}>
        <Text>Left</Text>
        {!leftActionActivated && (
          <TouchableOpacity
            style={[listsStyles.backRightBtn, listsStyles.backRightBtnLeft]}
            onPress={onClose}>
            <MaterialCommunityIcons
              name="archive-arrow-up-outline"
              size={25}
              style={listsStyles.trash}
              color="#fff"
            />
          </TouchableOpacity>
        )}
        {!leftActionActivated && (
          <Animated.View
            style={[
              listsStyles.backRightBtn,
              listsStyles.backRightBtnRight,
              {
                flex: 1,
                width: rowActionAnimatedValue,
              },
            ]}>
            <TouchableOpacity
              style={[listsStyles.backRightBtn, listsStyles.backRightBtnRight]}
              onPress={onDelete}>
              <Animated.View
                style={[
                  listsStyles.trash,
                  {
                    transform: [
                      {
                        scale: swipeAnimatedValue.interpolate({
                          inputRange: [-90, -45],
                          outputRange: [1, 0],
                          extrapolate: 'clamp',
                        }),
                      },
                    ],
                  },
                ]}>
                <MaterialCommunityIcons
                  name="trash-can-outline"
                  size={25}
                  color="#fff"
                />
              </Animated.View>
            </TouchableOpacity>
          </Animated.View>
        )}
      </Animated.View>
    );
  };

  const renderHiddenItem = (data, rowMap) => {
    const rowActionAnimatedValue = new Animated.Value(75);
    const rowHeightAnimatedValue = new Animated.Value(60);

    return (
      <HiddenItemWithActions
        data={data}
        rowMap={rowMap}
        rowActionAnimatedValue={rowActionAnimatedValue}
        rowHeightAnimatedValue={rowHeightAnimatedValue}
        onClose={() => unarchiveRow(rowMap, data.item.key, data.item)}
        onDelete={() => deleteRow(rowMap, data.item.key, data.item.id)}
      />
    );
  };

  return (
    <View style={listsStyles.container}>
      <SortListsSwitch sortType={sortType} setSortType={setSortType} />
      <StatusBar barStyle="dark-content" />
      <SwipeListView
        data={archivedShoppingLists}
        renderItem={renderItem}
        renderHiddenItem={renderHiddenItem}
        leftOpenValue={75}
        rightOpenValue={-150}
        disableRightSwipe
        onRowDidOpen={onRowDidOpen}
        leftActivationValue={100}
        rightActivationValue={-200}
        leftActionValue={0}
        rightActionValue={-500}
        onLeftAction={onLeftAction}
        onRightAction={onRightAction}
        onLeftActionStatusChange={onLeftActionStatusChange}
        onRightActionStatusChange={onRightActionStatusChange}
      />
    </View>
  );
};
