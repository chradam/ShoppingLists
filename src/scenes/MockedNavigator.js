import React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import {createStackNavigator} from '@react-navigation/stack';

const MockStack = createStackNavigator();

export default ({component, params = {}}) => (
  <NavigationContainer>
    <MockStack.Navigator>
      <MockStack.Screen
        name="MockedScreen"
        component={component}
        initialParams={params}
      />
    </MockStack.Navigator>
  </NavigationContainer>
);
