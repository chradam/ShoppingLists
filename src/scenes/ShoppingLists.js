import React, {useState, useCallback} from 'react';
import {
  View,
  Text,
  Animated,
  TouchableHighlight,
  TouchableOpacity,
  StatusBar,
} from 'react-native';
import {FAB} from 'react-native-elements';
import {SwipeListView} from 'react-native-swipe-list-view';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import {useFocusEffect} from '@react-navigation/native';

import {listsStyles} from '../styles';
import {LIST_MODE, LIST_STATUS} from '../config/Settings';
import model from '../model/Model';
import {sortList} from '../utils/index';
import {SortListsSwitch} from '../components/SortListsSwitch';

export const ShoppingLists = ({navigation}) => {
  const [shoppingLists, setShoppingLists] = useState(null);
  const [sortType, setSortType] = useState('desc');

  useFocusEffect(
    useCallback(() => {
      model.readShoppingLists(LIST_STATUS.active).then(lists => {
        const sortedLists = sortList(lists, sortType);

        setShoppingLists(
          sortedLists.map((list, index) => ({
            key: `${index}`,
            id: list.id,
            date: list.date,
            title: list.title,
            products: list.products,
            status: list.status,
          })),
        );
      });

      return () => {};
    }, [sortType]),
  );

  const closeRow = (rowMap, rowKey) => {
    if (rowMap[rowKey]) {
      rowMap[rowKey].closeRow();
    }
  };

  const archiveRow = (rowMap, rowKey, list) => {
    closeRow(rowMap, rowKey);
    const newData = [...shoppingLists];
    const prevIndex = shoppingLists.findIndex(item => item.key === rowKey);
    newData.splice(prevIndex, 1);
    setShoppingLists(newData);

    model.updateShoppingList(list, LIST_STATUS.archived);
  };

  const deleteRow = (rowMap, rowKey, itemId) => {
    model
      .deleteShoppingList(itemId)
      .then(() => {
        closeRow(rowMap, rowKey);
        const newData = [...shoppingLists];
        const prevIndex = shoppingLists.findIndex(item => item.key === rowKey);
        newData.splice(prevIndex, 1);
        setShoppingLists(newData);
      })
      .catch(err => {
        console.log(err);
      });
  };

  const onRowDidOpen = rowKey => {
    // console.log('This row opened', rowKey);
  };

  const onLeftActionStatusChange = rowKey => {
    console.log('onLeftActionStatusChange', rowKey);
  };

  const onRightActionStatusChange = rowKey => {
    console.log('onRightActionStatusChange', rowKey);
  };

  const onRightAction = rowKey => {
    console.log('onRightAction', rowKey);
  };

  const onLeftAction = rowKey => {
    console.log('onLeftAction', rowKey);
  };

  const VisibleItem = props => {
    const {
      data,
      rowHeightAnimatedValue,
      removeRow,
      leftActionState,
      rightActionState,
    } = props;

    if (rightActionState) {
      Animated.timing(rowHeightAnimatedValue, {
        toValue: 0,
        duration: 200,
        useNativeDriver: false,
      }).start(() => {
        removeRow();
      });
    }

    return (
      <Animated.View
        style={[listsStyles.rowFront, {height: rowHeightAnimatedValue}]}>
        <TouchableHighlight
          style={listsStyles.rowFrontVisible}
          onPress={() =>
            navigation.navigate('AddShoppingList', {
              name: 'Edit Shopping List',
              mode: LIST_MODE.edit,
              paramList: data.item,
            })
          }
          underlayColor={'#aaa'}>
          <View>
            <Text style={listsStyles.title} numberOfLines={1}>
              {data.item.title}
            </Text>
            <Text>
              {new Date(data.item.date).toLocaleString('pl-PL', {
                timeZone: 'Europe/Warsaw',
              })}
            </Text>
          </View>
        </TouchableHighlight>
      </Animated.View>
    );
  };

  const renderItem = (data, rowMap) => {
    const rowHeightAnimatedValue = new Animated.Value(60);

    return (
      <VisibleItem
        data={data}
        rowHeightAnimatedValue={rowHeightAnimatedValue}
        removeRow={() => deleteRow(rowMap, data.item.key, data.item.id)}
      />
    );
  };

  const HiddenItemWithActions = props => {
    const {
      swipeAnimatedValue,
      leftActionActivated,
      rightActionActivated,
      rowActionAnimatedValue,
      rowHeightAnimatedValue,
      onClose,
      onDelete,
    } = props;

    if (rightActionActivated) {
      Animated.spring(rowActionAnimatedValue, {
        toValue: 500,
        useNativeDriver: false,
      }).start();
    } else {
      Animated.spring(rowActionAnimatedValue, {
        toValue: 75,
        useNativeDriver: false,
      }).start();
    }

    return (
      <Animated.View
        style={[listsStyles.rowBack, {height: rowHeightAnimatedValue}]}>
        <Text>Left</Text>
        {!leftActionActivated && (
          <TouchableOpacity
            style={[listsStyles.backRightBtn, listsStyles.backRightBtnLeft]}
            onPress={onClose}>
            <MaterialCommunityIcons
              name="archive-arrow-down-outline"
              size={25}
              style={listsStyles.trash}
              color="#fff"
            />
          </TouchableOpacity>
        )}
        {!leftActionActivated && (
          <Animated.View
            style={[
              listsStyles.backRightBtn,
              listsStyles.backRightBtnRight,
              {
                flex: 1,
                width: rowActionAnimatedValue,
              },
            ]}>
            <TouchableOpacity
              style={[listsStyles.backRightBtn, listsStyles.backRightBtnRight]}
              onPress={onDelete}>
              <Animated.View
                style={[
                  listsStyles.trash,
                  {
                    transform: [
                      {
                        scale: swipeAnimatedValue.interpolate({
                          inputRange: [-90, -45],
                          outputRange: [1, 0],
                          extrapolate: 'clamp',
                        }),
                      },
                    ],
                  },
                ]}>
                <MaterialCommunityIcons
                  name="trash-can-outline"
                  size={25}
                  color="#fff"
                />
              </Animated.View>
            </TouchableOpacity>
          </Animated.View>
        )}
      </Animated.View>
    );
  };

  const renderHiddenItem = (data, rowMap) => {
    const rowActionAnimatedValue = new Animated.Value(75);
    const rowHeightAnimatedValue = new Animated.Value(60);

    return (
      <HiddenItemWithActions
        data={data}
        rowMap={rowMap}
        rowActionAnimatedValue={rowActionAnimatedValue}
        rowHeightAnimatedValue={rowHeightAnimatedValue}
        onClose={() => archiveRow(rowMap, data.item.key, data.item)}
        onDelete={() => deleteRow(rowMap, data.item.key, data.item.id)}
      />
    );
  };

  return (
    <View style={listsStyles.container}>
      <SortListsSwitch sortType={sortType} setSortType={setSortType} />
      <StatusBar barStyle="dark-content" />
      <SwipeListView
        data={shoppingLists}
        renderItem={renderItem}
        renderHiddenItem={renderHiddenItem}
        leftOpenValue={75}
        rightOpenValue={-150}
        disableRightSwipe
        onRowDidOpen={onRowDidOpen}
        leftActivationValue={100}
        rightActivationValue={-200}
        leftActionValue={0}
        rightActionValue={-500}
        onLeftAction={onLeftAction}
        onRightAction={onRightAction}
        onLeftActionStatusChange={onLeftActionStatusChange}
        onRightActionStatusChange={onRightActionStatusChange}
      />
      <FAB
        placement="right"
        size="large"
        color="deepskyblue"
        icon={{name: 'add', type: 'ionicons', color: 'white'}}
        onPress={() =>
          navigation.navigate('AddShoppingList', {name: 'Add Shopping List'})
        }
      />
    </View>
  );
};
