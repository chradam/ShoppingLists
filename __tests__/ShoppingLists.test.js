import React from 'react';
import renderer from 'react-test-renderer';
import MockedNavigator from '../src/scenes/MockedNavigator';
import {ShoppingLists} from '../src/scenes/ShoppingLists';
import Colors from '../src/constants/Colors';

test('render', () => {
  const snap = renderer.create(
    <MockedNavigator
      component={ShoppingLists}
      initialParams={{
        title: 'Shopping Lists',
        headerStyle: {
          backgroundColor: Colors.primary,
        },
        headerTintColor: '#fff',
      }}
    />,
  );
  expect(snap).toMatchSnapshot();
});
