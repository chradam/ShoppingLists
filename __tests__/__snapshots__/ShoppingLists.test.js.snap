// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Object {
          "zIndex": 1,
        }
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        onLayout={[Function]}
        pointerEvents="box-none"
        style={null}
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 0,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgb(255, 255, 255)",
                "borderBottomColor": "rgb(216, 216, 216)",
                "flex": 1,
                "shadowColor": "rgb(216, 216, 216)",
                "shadowOffset": Object {
                  "height": 0.5,
                  "width": 0,
                },
                "shadowOpacity": 0.85,
                "shadowRadius": 0,
              }
            }
          />
        </View>
        <View
          pointerEvents="box-none"
          style={
            Object {
              "height": 64,
              "maxHeight": undefined,
              "minHeight": undefined,
              "opacity": undefined,
              "transform": undefined,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "height": 20,
              }
            }
          />
          <View
            pointerEvents="box-none"
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "marginHorizontal": 16,
                  "opacity": 1,
                }
              }
            >
              <Text
                accessibilityRole="header"
                aria-level="1"
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 17,
                    "fontWeight": "600",
                  }
                }
              >
                MockedScreen
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <RNSScreenContainer
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <RNSScreen
        activityState={2}
        forwardedRef={[Function]}
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "opacity": 1,
            }
          }
        />
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onGestureEnd={[Function]}
          onOpen={[Function]}
          onTransition={[Function]}
          pointerEvents="box-none"
          style={
            Array [
              Object {
                "overflow": undefined,
              },
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
            ]
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            needsOffscreenAlphaCompositing={false}
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "backgroundColor": "#f4f4f4",
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        accessibilityLabel={null}
                        style={
                          Array [
                            Object {
                              "flexDirection": "row",
                            },
                            Object {},
                          ]
                        }
                        testID={null}
                      >
                        <View
                          onMoveShouldSetResponder={[Function]}
                          onMoveShouldSetResponderCapture={[Function]}
                          onResponderEnd={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderReject={[Function]}
                          onResponderRelease={[Function]}
                          onResponderStart={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          onStartShouldSetResponderCapture={[Function]}
                          style={
                            Object {
                              "flex": 1,
                            }
                          }
                        >
                          <View
                            onLayout={[Function]}
                            style={
                              Object {
                                "backgroundColor": "#FFFFFF",
                                "borderRadius": 50,
                                "height": 40,
                              }
                            }
                          >
                            <View
                              style={
                                Object {
                                  "alignItems": "center",
                                  "borderColor": "#C9C9C9",
                                  "borderRadius": 50,
                                  "borderWidth": 1,
                                  "flex": 1,
                                  "flexDirection": "row",
                                }
                              }
                            >
                              <View
                                accessible={true}
                                focusable={true}
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  Object {
                                    "alignItems": "center",
                                    "flex": 1,
                                    "flexDirection": "row",
                                    "justifyContent": "center",
                                    "opacity": 1,
                                  }
                                }
                              >
                                <Text
                                  allowFontScaling={false}
                                  style={
                                    Array [
                                      Object {
                                        "color": "#fff",
                                        "fontSize": 25,
                                      },
                                      undefined,
                                      Object {
                                        "fontFamily": "Material Design Icons",
                                        "fontStyle": "normal",
                                        "fontWeight": "normal",
                                      },
                                      Object {},
                                    ]
                                  }
                                >
                                  󰒽
                                </Text>
                                <Text
                                  style={
                                    Array [
                                      Object {
                                        "backgroundColor": "transparent",
                                        "color": "#FFFFFF",
                                        "fontSize": 14,
                                        "fontWeight": "normal",
                                        "textAlign": "center",
                                      },
                                      Object {},
                                    ]
                                  }
                                >
                                  DESC
                                </Text>
                              </View>
                              <View
                                accessible={true}
                                focusable={true}
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  Object {
                                    "alignItems": "center",
                                    "flex": 1,
                                    "flexDirection": "row",
                                    "justifyContent": "center",
                                    "opacity": 1,
                                  }
                                }
                              >
                                <Text
                                  allowFontScaling={false}
                                  style={
                                    Array [
                                      Object {
                                        "color": "#000",
                                        "fontSize": 25,
                                      },
                                      undefined,
                                      Object {
                                        "fontFamily": "Material Design Icons",
                                        "fontStyle": "normal",
                                        "fontWeight": "normal",
                                      },
                                      Object {},
                                    ]
                                  }
                                >
                                  󰒼
                                </Text>
                                <Text
                                  style={
                                    Array [
                                      Object {
                                        "backgroundColor": "transparent",
                                        "color": "#000000",
                                        "fontSize": 14,
                                        "fontWeight": "normal",
                                        "textAlign": "center",
                                      },
                                      Object {},
                                    ]
                                  }
                                >
                                  ASC
                                </Text>
                              </View>
                            </View>
                          </View>
                        </View>
                      </View>
                      <RCTScrollView
                        closeOnRowBeginSwipe={false}
                        closeOnRowOpen={true}
                        closeOnRowPress={true}
                        closeOnScroll={true}
                        data={null}
                        directionalDistanceChangeThreshold={2}
                        disableHiddenLayoutCalculation={false}
                        disableLeftSwipe={false}
                        disableRightSwipe={true}
                        disableVirtualization={false}
                        getItem={[Function]}
                        getItemCount={[Function]}
                        horizontal={false}
                        initialNumToRender={10}
                        keyExtractor={[Function]}
                        leftActionValue={0}
                        leftActivationValue={100}
                        leftOpenValue={75}
                        maxToRenderPerBatch={10}
                        onContentSizeChange={[Function]}
                        onEndReachedThreshold={2}
                        onLayout={[Function]}
                        onLeftAction={[Function]}
                        onLeftActionStatusChange={[Function]}
                        onMomentumScrollEnd={[Function]}
                        onRightAction={[Function]}
                        onRightActionStatusChange={[Function]}
                        onRowDidOpen={[Function]}
                        onScroll={[Function]}
                        onScrollBeginDrag={[Function]}
                        onScrollEndDrag={[Function]}
                        previewFirstRow={false}
                        previewRepeat={false}
                        previewRepeatDelay={1000}
                        recalculateHiddenLayout={false}
                        removeClippedSubviews={false}
                        renderHiddenItem={[Function]}
                        renderItem={[Function]}
                        rightActionValue={-500}
                        rightActivationValue={-200}
                        rightOpenValue={-150}
                        scrollEventThrottle={50}
                        stickyHeaderIndices={Array []}
                        swipeToClosePercent={50}
                        swipeToOpenPercent={50}
                        swipeToOpenVelocityContribution={0}
                        updateCellsBatchingPeriod={50}
                        useAnimatedList={false}
                        useNativeDriver={true}
                        viewabilityConfigCallbackPairs={Array []}
                        windowSize={21}
                      >
                        <View />
                      </RCTScrollView>
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "borderRadius": 28,
                            "bottom": 0,
                            "flexDirection": "row",
                            "justifyContent": "center",
                            "margin": 16,
                            "opacity": 1,
                            "position": "absolute",
                            "right": 0,
                            "transform": Array [
                              Object {
                                "scale": 1,
                              },
                            ],
                          }
                        }
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "borderRadius": 3,
                                "overflow": "hidden",
                              },
                              Object {
                                "borderRadius": 3,
                              },
                              Object {
                                "borderRadius": 28,
                                "elevation": 4,
                              },
                              false,
                            ]
                          }
                        >
                          <View
                            accessibilityRole="button"
                            accessibilityState={
                              Object {
                                "busy": false,
                                "disabled": false,
                              }
                            }
                            accessible={true}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "opacity": 1,
                              }
                            }
                          >
                            <View
                              style={
                                Object {
                                  "alignItems": "center",
                                  "backgroundColor": "deepskyblue",
                                  "borderColor": "#2089dc",
                                  "borderRadius": 3,
                                  "borderWidth": 0,
                                  "flexDirection": "row",
                                  "height": 56,
                                  "justifyContent": "center",
                                  "padding": 16,
                                  "width": 56,
                                }
                              }
                            >
                              <View
                                style={
                                  Object {
                                    "height": 56,
                                    "marginHorizontal": 5,
                                    "overflow": "hidden",
                                    "padding": 16,
                                    "width": 56,
                                  }
                                }
                              >
                                <View>
                                  <View
                                    style={
                                      Object {
                                        "alignItems": "center",
                                        "backgroundColor": "transparent",
                                        "justifyContent": "center",
                                      }
                                    }
                                  >
                                    <Text
                                      allowFontScaling={false}
                                      style={
                                        Array [
                                          Object {
                                            "color": "white",
                                            "fontSize": 24,
                                          },
                                          Object {
                                            "backgroundColor": "transparent",
                                          },
                                          Object {
                                            "fontFamily": "Material Icons",
                                            "fontStyle": "normal",
                                            "fontWeight": "normal",
                                          },
                                          Object {},
                                        ]
                                      }
                                      testID="iconIcon"
                                    >
                                      
                                    </Text>
                                  </View>
                                </View>
                              </View>
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RNSScreen>
    </RNSScreenContainer>
  </RNCSafeAreaProvider>
</View>
`;
